web/templatetags/markdown_filters.py

```
@@ -1,11 +1,68 @@
+
+    Uses bleach to sanitize the HTML output from markdownify,
+    preventing XSS attacks while preserving safe markdown formatting.
+    """
+    html = markdownify(text)
+    clean_html = bleach.clean(
+        html,
+        tags=ALLOWED_TAGS,
+        attributes=ALLOWED_ATTRIBUTES,
+        strip=True,
+    )
+    return mark_safe(clean_html)
```

Ruff (0.15.0)

[error] Use of `mark_safe` may expose cross-site scripting vulnerabilities

(S308)